addons/redis/redis-cluster-scripts/redis-cluster-server-start.sh

```diff
@@ -377,19 +377,37 @@ scale_redis_cluster_replica() {
     replicated_command="redis-cli --cluster add-node $current_node_with_port $primary_node_endpoint_with_port --cluster-slave --cluster-master-id $primary_node_cluster_id -a $REDIS_DEFAULT_PASSWORD"
   fi
   echo "scale out replica replicated command: $replicated_command" | sed "s/$REDIS_DEFAULT_PASSWORD/********/g"
-  replicated_output=$($replicated_command)
-  replicated_exit_code=$?
-  set -x
-  echo "Scale out replica replicated command result: $replicated_output"
-  if [ $replicated_exit_code -ne 0 ]; then
+  # Keep trying until add-node is done.
+  # And the current_pod_fqdn DNS cache time is normally 30 seconds, thus make 30 attempts
+  local max_attempts=30
+  local attempt=1
+
+  while [ $attempt -le $max_attempts ]
```

Comment:
有 while 循环的 shell 脚本最好加一下对 SIGTERM 的信号响应：

// 捕获信号
trap "exit_handler" SIGTERM SIGINT

exit_handler() {
    echo "Received stop signal, cleaning up..."
    do some clean job
    exit 0
}

Reply:
In this file, there is a retry() function, which has an until loop in it. Also, a while true loop is just behind the code I submitted. So a global `exit_handler` is needed for this script file?

BTW: Because the retry function just validates the return code, but not checking the error msgs like the "is not empty", thus the retry() function was not used by my code.

Reply:
add a global one to make the scripts stop and exit ASAP

Reply:
> In this file, there is a retry() function, which has an until loop in it. Also, a while true loop is just behind the code I submitted. So a global `exit_handler` is needed for this script file?
> 
> BTW: Because the retry function just validates the return code, but not checking the error msgs like the "is not empty", thus the retry() function was not used by my code.

I think it could be considered in later pr.

Reply:
OK. Let's fix issues one by one.